src/timer/timer_game_calendar.cpp

```diff
@@ -93,28 +93,33 @@ void TimeoutTimer<TimerGameCalendar>::Elapsed(TimerGameCalendar::TElapsed trigge
 	}
 }
 
+/**
+ * Every-tick handler for possibly advancing calendar time.
+ * @note The return value is used to inform other functions whether we advanced TimerGameCalendar::date_fract.
+ * @return True if we advanced TimerGameCalendar::date_fract, otherwise false if we didn't change anything or only advanced TimerGameCalendar::sub_date_fract.
```

Comment:
Here too (in combination with the other comments of mine). You now appear to assign a different return "meaning" for each class type. That is not ideal.

Better just make the return value mean something in one place, and not something different for each implementation of the template. Basically, there shouldn't be a need to have a comment per template implementation, and the comment at the definition should explain what is going on.